File: Easy/lc202.py
# Happy Number
class Solution(object):
    def isHappy(self, n):
        seen=set()
        while n!=1:
            if n in seen:     
                return False
            seen.add(n)
            s=str(n)
            count=0
            for i in s:
                count+=int(i)**2
            n=count     
        return  True             
s=Solution()
print(s.isHappy(19))  # Output: true
print(s.isHappy(2))   # Output: false
print(s.isHappy(7))   # Output: true
        

File: Easy/lc202.java
import java.util.HashSet;
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> seen = new HashSet<>();
        while(n!=1){
            if(seen.contains(n)){
                return false;
            }
            seen.add(n);
            int c=0;
            while(n>0){
                int temp=n%10;
                c+=temp*temp;
                n=n/10;
            }
            n=c;
        }
        return true;
    }
    public static void main(String[] args) {
        Solution s=new Solution();
        System.out.println(s.isHappy(19));
    }
}